### ISSUE TITLE
Front End / Back End Separate Solution 

### ISSUE THREAD

**Issue by StyleTang** [apache/rocketmq-dashboard#10]
#### Please describe the feature you are requesting.

Now the code of RocketMQ Dashboard mixed the front-end and back-end code together, the front-end architecture is too old, it is hard to maintain, for example, front end package is difficult to manage, the fancy features of the front-end framework cannot be used, can not add front end unit test etc...

We hope to separate the front-end and back-end modules. The front-end choose **React**, uses npm to manage package, and the back-end continues to use **Spring-boot** and maven.

This may introduce the deployment problem of front-end static resources. At present, we propose to use maven plugin to handle it. During the spring-boot packaging process, maven plugin should compile and package the front-end code and put it into spring-boot.jar. When deploying, only the spring-boot server needs to be deployed (It will contains the front end resource). 

#### Provide any additional detail on your proposed use case for this feature.

We propose the rocketmq-dashboard package should like this.

rocketmq-dashboard

|-------src ( backend: **Spring-boot**)

|-------frontend (**React**)

When we execute package cmd, maven plugin should auto compile and release the front end resource, and build them into rocketmq-dashboard.jar

```shell
mvn clean package -Dmaven.test.skip=true
```

then we can run it just like a ordinary spring boot server.

```
java -jar target/rocketmq-dashboard.jar
```

#### Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

**blocker**. (Architecture upgrade with separation of front and back ends strong dependence on it)

#### If there are some sub-tasks using -[] for each subtask and create a corresponding issue to map to the sub task:

This is related to the front-end architecture selection, but it can be done in parallel.
